src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -142,7 +142,9 @@ internal void SetBuffer(byte[] buffer, int offset, int count)
 
         internal bool TryGetBuffer(out ArraySegment<byte> buffer)
         {
-            return _bufferedContent != null && _bufferedContent.TryGetBuffer(out buffer);
+            bool result = _bufferedContent != null && _bufferedContent.TryGetBuffer(out buffer);
+            Debug.Assert(result, "Performance: TryGetBuffer failed, which means that reads from the content will copy the data!");
```

Comment:
Anyway, since this PR is about adding an assert and that assert is undesirable, I'm closing the PR. If logging is needed, it can be added in a PR dedicated to that. Thanks.